.envrc

```diff
@@ -9,21 +9,21 @@
 # DO NOT STORE SECRETS HERE - this file is version-controlled! You should store secrets in a `.secrets` file, which is
 # not version-controlled - this can then be sourced here, using `source_env ".secrets"`.
 
+# Extract the variables to `.env` if required. Note `.env` is NOT version-controlled
+sed -n 's/^export \(.*\)$/\1/p' .envrc | sed -e 's?$(pwd)?'"$(pwd)"'?g' > .env
+
 # Add the working directory to PYTHONPATH; allows Jupyter notebooks in the `notebooks` folder to import `src`
 export PYTHONPATH="$PYTHONPATH:$(pwd)"
 
 # Import secrets from an untracked file `.secrets`
-# source_env ".secrets"
-
-# Add project directory to enviro vars
-export PROJECT_DIR=$(pwd)
+source_env ".secrets"
 
 # Add environment variables for the `data` directories
 export DIR_DATA=$(pwd)/data
-export DIR_DATA_EXTERNAL=$DIR_DATA/external
-export DIR_DATA_RAW=$DIR_DATA/raw
-export DIR_DATA_INTERIM=$DIR_DATA/interim
-export DIR_DATA_PROCESSED=$DIR_DATA/processed
+export DIR_DATA_EXTERNAL=$(pwd)/data/external
```

Comment:
why these changes? So that an env variable is not dependent on another?

Reply:
Yup, I think the `.env` file creation treats `$DIR_DATA` literally (or is static) so if we use `$(pwd)` instead, then that should be dynamic according to what your `pwd` is.